docs/notebooks/active_learning_for_binary_classification.pct.py

```diff
@@ -0,0 +1,179 @@
+
+# %% [markdown]
+# For the binary classification model, we use the Variational Gaussian Process with Bernoulli likelihood. For more detail of this model, see <cite data-cite="Nickisch08a">[Nickisch et al.](https://www.jmlr.org/papers/volume9/nickisch08a/nickisch08a.pdf)</cite>. Here we use trieste's gpflow model builder `build_vgp_classifier`.
+# User can also use Sparse Variational Gaussian Process(SVGP) for building the classification model via `build_svgp` function and `SparseVariational` class. SVGP is preferable for bigger amount of data.
+
+# %%
+from trieste.models.gpflow import VariationalGaussianProcess
+from trieste.models.gpflow.builders import build_vgp_classifier
+
+model = VariationalGaussianProcess(
+    build_vgp_classifier(initial_data, search_space, noise_free=True)
+)
+
+# %% [markdown]
+# Lets see our model landscape using only those initial data
+
+# %%
+model.update(initial_data)
+model.optimize(initial_data)
+
+mean, variance = model.predict(query_points)
+
+plt.figure()
+plt.contourf(*grid_query_points, np.reshape(mean, [density] * input_dim))
+plt.plot(
+    initial_data.query_points[:, 0],
+    initial_data.query_points[:, 1],
+    "ko",
+    markersize=10,
+)
+plt.title("Mean")
+plt.colorbar()
+plt.show()
+
+plt.figure()
+plt.contourf(*grid_query_points, np.reshape(variance, [density] * input_dim))
+plt.colorbar()
+plt.plot(
+    initial_data.query_points[:, 0],
+    initial_data.query_points[:, 1],
+    "ko",
+    markersize=10,
+)
+plt.title("Variance")
+plt.show()
+
+
+# %% [markdown]
+# ## The acquisition process
+#
+# We can construct the BALD acqusition function which maximise information gain about the model parameters, by maximising the mutual information between predictions and model posterior:
+#
+# $$\mathbb{I}\left[y, \boldsymbol{\theta} \mid \mathbf{x}, \mathcal{D}\right]=\mathbb{H}\left[y \mid \mathbf{x}, \mathcal{D}\right]-\mathbb{E}_{p\left(\boldsymbol{\theta} \mid \mathcal{D}\right)}[\mathbb{H}[y \mid \mathbf{x}, \boldsymbol{\theta}]]$$
+#
+# See <cite data-cite="houlsby2011bayesian">[Houlsby et al.](https://arxiv.org/pdf/1112.5745.pdf)</cite> for more details. Then, Trieste's `EfficientGlobalOptimization` is used for the query rule:
+
+# %%
+initial_models = model
+acq = BayesianActiveLearningByDisagreement()
+rule = trieste.acquisition.rule.EfficientGlobalOptimization(acq)  # type: ignore
+
+# %% [markdown]
+# ## Run the active learning loop
+# Let's run our active learning iteration:
+
+# %%
+n_steps = 30
+bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+results = bo.optimize(n_steps, initial_data, initial_models, rule, track_state=False)
+final_dataset = results.try_get_final_datasets()[OBJECTIVE]
+final_model = results.try_get_final_models()[OBJECTIVE]
+
+# %% [markdown]
+# ## Visualising the result
+# Now, we can visualize our model after the active learning run
+
+# %% Plot BO results
+mean, variance = final_model.predict(query_points)
+
+
+def invlink(f):
```

Comment:
since you use it only once immediately below perhaps there is no need to define a function, just compute it directly below